question_answering_service/__init__.py

```diff
@@ -31,19 +32,27 @@ def create_app():
             "client_secret": os.getenv("CLIENT_SECRET"),
             "refresh_token": os.getenv("REFRESH_TOKEN"),
         })
-
         gsclient = gspread.authorize(credentials)
 
     except Exception as e:
         logging.error("Error authorizing google sheets client : " + str(e), exc_info=True)
-        gsclient = None
+        sys.exit(1)
 
     logging.info("Reading documents from google sheet")
-    data = read_documents(gsclient, "1h0WX6lo_aCtnAo1dftVtFqR4CqFWzY-0kT2ZWfCSrWg", "Sheet1")
+    data = read_documents(gsclient, os.getenv("SHEET_ID"), os.getenv("SHEET_NAME"))
+
+    if not data:
+        sys.exit(1)
 
     for item in data:
+        content_embedding = get_embedding(item['content'])
```

Comment:
One way to do this might be raise exception inside `read_documents` and `get_embedding`. Then we don't need to check and do sys.exit in each place separately.  If any of them fails then we do sys.exit

Reply:
Removed the try - except blocks in `read_documents` and `get_embedding` and enclosed the function calling in a try except block in this file.